toolchain/check/action.cpp

```diff
@@ -98,14 +100,14 @@ static auto AddDependentActionSpliceImpl(Context& context,
 // their concrete values, so that the action doesn't need to know which specific
 // it is operating on.
 static auto RefineOperand(Context& context, SemIR::LocId loc_id,
-                          SemIR::IdKind kind, int32_t arg) -> int32_t {
-  if (kind == SemIR::IdKind::For<SemIR::MetaInstId>) {
-    auto inst_id = SemIR::MetaInstId(arg);
+                          SemIR::Inst::ArgAndKind arg) -> int32_t {
+  if (arg.kind == SemIR::IdKind::For<SemIR::MetaInstId>) {
+    auto inst_id = arg.As<SemIR::MetaInstId>();
```

Comment:
This suggests a `TryAs` function would be useful:
```suggestion
  if (auto inst_id = arg.TryAs<SemIR::MetaInstId>()) {
```

Reply:
Done